src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors: Stefan Irimescu, Can Berker Cikis
+ *
+ */
+
+package org.rumbledb.compiler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.rumbledb.api.Item;
+import org.rumbledb.config.RumbleRuntimeConfiguration;
+import org.rumbledb.exceptions.UnexpectedTypeException;
+import org.rumbledb.expressions.AbstractNodeVisitor;
+import org.rumbledb.expressions.Node;
+import org.rumbledb.expressions.module.VariableDeclaration;
+import org.rumbledb.items.ItemFactory;
+import org.rumbledb.types.SequenceType.Arity;
+
+import sparksoniq.semantics.DynamicContext;
+
+/**
+ * Static context visitor implements a multi-pass algorithm that enables function hoisting
```

Comment:
This comment should be removed. Feel free to add a documentation for the DynamicContextVisitor's purpose as well.

Reply:
Done